src/main/java/com/google/devtools/build/lib/remote/RemoteSpawnRunner.java

```diff
@@ -322,9 +321,20 @@ public SpawnResult exec(Spawn spawn, SpawnExecutionContext context)
                 spawnMetrics.setUploadTime(
                     uploadTime.elapsed().minus(networkTime.getDuration().minus(networkTimeStart)));
               }
+              final AtomicBoolean reportedExecuting =
+                  new AtomicBoolean(false);
               ExecuteResponse reply;
               try (SilentCloseable c = prof.profile(REMOTE_EXECUTION, "execute remotely")) {
-                reply = remoteExecutor.executeRemotely(request);
+                reply = remoteExecutor.executeRemotely(request, (operation) -> {
+                  if (!reportedExecuting.get()) {
```

Comment:
This is technically a check-then-act race. It is probably not a problem here because the caller is single-threaded (although then, why use AtomicBoolean), but IMO you should get in a habit of using compareAndSet rather than separate get and set calls.